drivers/spi/spi_rpi_pico_pio.c

```diff
@@ -738,6 +735,8 @@ int spi_pico_pio_init(const struct device *dev)
 		return rc;
 	}
 
+        data->pio_sm = dev_cfg->pio_sm;
```

Comment:
This is really awful, but I would also like to have `pio` to stay inside `config` instead of `data`...

Reply:
Let me fix this in another PR.

Reply:
Done in #85167.